server/src/test/java/com/decathlon/ara/cartography/SquashExporterTest.java

```diff
@@ -0,0 +1,122 @@
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SquashExporterTest {
+
+    @Spy
+    private SquashExporter sut;
+
+    @Test
+    public void getName() {
+        // Given
+        final String expected = "SquashTM";
+        // When
+        String actualName = this.sut.getName();
+        // Then
+        Assertions.assertThat(actualName).isEqualTo(expected);
+    }
+
+    @Test
+    public void getDescription() {
+        // Given
+        final String expected = "Export this cartography to import it as requirements in SquashTM";
+        // When
+        String actualName = this.sut.getDescription();
+        // Then
+        Assertions.assertThat(actualName).isEqualTo(expected);
+    }
+
+    @Test
+    public void getFormat() {
+        // Given
+        final String expected = "xls";
+        // When
+        String actualName = this.sut.getFormat();
+        // Then
+        Assertions.assertThat(actualName).isEqualTo(expected);
+    }
+
+    @Test
+    public void generate_should_create_a_header_row_even_when_no_functionalities() {
+        // Given
+        List<FunctionalityDTO> functionalities = new ArrayList<>();
+        Map<String, String> requiredInfos = new HashMap<>();
+        requiredInfos.put(SquashExporter.PROJECT_NAME, "project");
+        requiredInfos.put(SquashExporter.USER, "user");
+        // When
+        byte[] result = this.sut.generate(functionalities, requiredInfos);
+        // Then
+        Mockito.verify(this.sut, Mockito.times(1)).createHeaderRow(Mockito.any());
+        Mockito.verify(this.sut, Mockito.never()).addRowToSheet(Mockito.any(), Mockito.anyInt(), Mockito.anyString(),
+                Mockito.anyString(), Mockito.anyString(), Mockito.anyString(), Mockito.any(), Mockito.anyString());
+        Assertions.assertThat(result).isNotEmpty();
+    }
+
+
+    @Test
+    public void generate_should_generate_one_line_per_functionality() {
+        // Given
+        List<FunctionalityDTO> functionalities = new ArrayList<>();
+        functionalities.add(this.create_dummy_functionality(1));
+        functionalities.add(this.create_dummy_functionality(2));
+        Map<String, String> requiredInfos = new HashMap<>();
+        requiredInfos.put(SquashExporter.PROJECT_NAME, "project");
+        requiredInfos.put(SquashExporter.USER, "user");
+        // When
+        byte[] result = this.sut.generate(functionalities, requiredInfos);
+        // Then
+        Mockito.verify(this.sut, Mockito.times(1)).createHeaderRow(Mockito.any());
+        Mockito.verify(this.sut, Mockito.times(functionalities.size())).addRowToSheet(Mockito.any(), Mockito.anyInt(), Mockito.anyString(),
+                Mockito.anyString(), Mockito.anyString(), Mockito.anyString(), Mockito.any(), Mockito.anyString());
+        Assertions.assertThat(result).isNotEmpty();
+    }
+
+    @Test
+    public void listRequiredFields() {
```

Comment:
if you write your test methods names with snake_case, you should do the same here as well

Reply:
The lack of snake case here is because I only test what the documentation of the `listRequiredFields` method tells, the "passing case", there is only one path to test in this code, so I don't need to differentiate the test cases